core/imodel-bridge/src/BadgersIssueReporter.ts

```diff
@@ -0,0 +1,283 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { DbResult, Guid, GuidString, Logger } from "@bentley/bentleyjs-core";
+import { ECDb } from "@bentley/imodeljs-backend";
+import { BridgeIssueReporter } from "./BridgeIssueReporter"
+import * as fs from "fs";
+import * as path from "path";
+import * as os from "os";
+import { BridgeLoggerCategory } from "./BridgeLoggerCategory";
+
+interface ContextInfo {
+  activityId: string;
+  contextId: string;
+  iModelId: string;
+  briefcaseId?: number;
+  operationType?: string;
+  jobId?: string;
+}
+
+interface FileInfo {
+  fileId?: string;
+  filePath?: string;
+}
+
+export interface PostPublishingProcessor {
+  (reportPath: string): Promise<void> | void;
+}
+
+/** BADGERS issue reporter. Can be provided to BridgeRunner in setIssueReporter
+ * @beta
+ */
+export class BadgersIssueReporter implements BridgeIssueReporter {
+  private _reportDb;
+  private _reportFile;
+  private _outputDir;
+  private _contextInfo: ContextInfo;
+  private _fileInfo: FileInfo;
+  private _postPublishingProcessor: PostPublishingProcessor;
+  private _dbPath: string = "";
+  private _deleteBadgersDb;
+
+  private TEMP_REPORT_FILE_RECORDS_TABLE = "BadgersReportFileRecords";
+  private TEMP_REPORT_AUDIT_RECORDS_TABLE = "BadgersReportAuditRecords";
+  private SRCFILE_TEMP_BRIDGE_ISSUES_TABLE = "BadgersSrcFileReports";
+  private TEMP_IGNORED_ELEM_RECORD_TABLE = "BadgersReportIgnoredElementRecords";
+
+  public constructor(contextId: string, iModelId: GuidString, activityId: string, fileId: string, outputDir?: any, operationType?: string,
```

Comment:
Can we constrain the possible values for `operationType` (vs. a freeform string)?